addons/hr_holidays/models/hr_leave.py

```diff
@@ -718,25 +721,23 @@ def name_get(self):
                                 person=target,
                                 leave_type=leave.holiday_status_id.name,
                                 duration=leave.number_of_hours_display,
-                                date=fields.Date.to_string(leave.date_from),
+                                date=fields.Date.to_string(date_from_utc) or "",
                             )
                         ))
                     else:
-                        user_tz = timezone(leave.tz)
-                        date_from_utc = leave.date_from.astimezone(user_tz).date()
                         res.append((
                             leave.id,
                             _("%(person)s on %(leave_type)s: %(duration).2f hours on %(date)s",
                                 person=target,
                                 leave_type=leave.holiday_status_id.name,
                                 duration=leave.number_of_hours_display,
-                                date=fields.Date.to_string(date_from_utc),
+                                date=fields.Date.to_string(date_from_utc) or "",
                             )
                         ))
                 else:
-                    display_date = fields.Date.to_string(leave.date_from)
-                    if leave.number_of_days > 1:
-                        display_date += ' / %s' % fields.Date.to_string(leave.date_to)
+                    display_date = fields.Date.to_string(date_from_utc) or ""
+                    if leave.number_of_days > 1 and date_from_utc:
+                        display_date += ' / %s' % fields.Date.to_string(date_to_utc) or ""
```

Comment:
Maybe something like this to avoid a "/" with nothing after
```suggestion
                    display_date = fields.Date.to_string(date_from_utc) or ""
                    date_to_string = fields.Date.to_string(date_to_utc) or False
                    if leave.number_of_days > 1 and date_from_utc and date_to_string:
                        display_date += ' / %s' % date_to_string
```